### DISCUSSION TITLE
Convert from OrderedDict?

### DISCUSSION THREAD

**Discussion by jprawiharjo** [omry/omegaconf#1155] [Q&A]
I have a nested dict that contains `OrderedDict`. How do I convert this to omegaconf? Both `save` and `create` complain:

`omegaconf.errors.UnsupportedValueType: Value 'OrderedDict' is not a supported primitive type`

**Comment by odelalleau** [comment]
Indeed, `OrderedDict` is currently not supported. Simplest solution is to convert to regular `dict`, e.g. with something like

```python
def to_regular_dict(container):
    if isinstance(container, dict):
        return {k: to_regular_dict(v) for k, v in container.items()}
    elif isinstance(container, list):
        return [to_regular_dict(k) for k in container]
    else:
        return container
```

**Reply by jprawiharjo** [reply]
Thanks. Although we can convert to omegaconf using this method, this removes the order of the dictionary keys, so I wouldn't mark this as an answer. Ideally, omegaconf handles ordered dict by converting it to `DictConfig` recursively.

**Reply by odelalleau** [reply]
This shouldn't change the order of the keys -- a regular Python `dict` preserves order as part of the standard since Python 3.7.
